### ISSUE TITLE
1.1.0 scaling behavior issue

### ISSUE THREAD

**Issue by papertigify** [taodo2291/xgesture_flutter#16]
Hello! First of all, thanks for your great package.<br>
I have an issue after upgrading to 1.1.0.<br>
I use onScaleStart, onScaleUpdate, onScaleEnd callbacks, and it seems like onScaleUpdate does not work properly (and sometimes I get the error, stack trace below). The other thing is, onScaleEnd is not called at all.
<details>
  <summary>Stack trace</summary>
  
```

The following RangeError was thrown while dispatching a pointer event:
RangeError (index): Invalid value: Only valid value is 0: 1
When the exception was thrown, this was the stack: 
#0      List.[] (dart:core-patch/growable_array.dart:281:36)
#1      _XGestureDetectorState.onPointerMove (package:gesture_x_detector/gesture_x_detector.dart:181:65)
#2      RenderPointerListener.handleEvent (package:flutter/src/rendering/proxy_box.dart:2892:29)
#3      GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:419:22)
#4      RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:322:11)
#5      GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:374:7)
#6      GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:338:5)
#7      GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:296:7)
#8      GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:279:7)
#12     _invoke1 (dart:ui/hooks.dart:170:10)
#13     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:331:7)
#14     _dispatchPointerDataPacket (dart:ui/hooks.dart:94:31)
(elided 3 frames from dart:async)
Event: PointerMoveEvent#b33dd(position: Offset(185.1, 510.2))
  position: Offset(185.1, 510.2)
Target: RenderPointerListener#5c4b8 relayoutBoundary=up11
  needs compositing
  parentData: <none> (can use size)
  constraints: BoxConstraints(0.0<=w<=392.7, 0.0<=h<=600.4)
  size: Size(392.7, 600.4)
  behavior: deferToChild
  listeners: down, move, up, cancel

  ```
  
</details>
  
<details>
  <summary>Logs</summary>
  
  ```
  
  [✓] Flutter (Channel stable, 2.10.5, on macOS 11.5.2 20G95 darwin-arm, locale en)
    • Flutter version 2.10.5 at /Users/papertigify/tools/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5464c5bac7 (10 days ago), 2022-04-18 09:55:37 -0700
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/papertigify/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] Connected device (2 available)
    • iPhone 11 Pro (mobile) • ios • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)• chrome • web-javascript • Google Chrome 100.0.4896.127

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
  
  ```

</details>

**Comment by taodo2291** [comment]
New version released: https://pub.dev/packages/gesture_x_detector/versions/1.1.1
Please let me know if it solved this issue.

**Comment by taodo2291** [comment]
Solved at [version 1.1.1](https://pub.dev/packages/gesture_x_detector/versions/1.1.1)
